src/wasapi.c

```diff
@@ -705,7 +713,9 @@ static int refresh_devices(struct SoundIoPrivate *si) {
         }
         if (FAILED(hr = IMMDevice_GetId(rd.default_capture_device, &rd.lpwstr))) {
             deinit_refresh_devices(&rd);
-            return SoundIoErrorOpeningDevice;
+            // MSDN states the IMMDevice_GetId can fail if the device is NULL, or if we're out of memory
+            // We know the device point isn't NULL so we're necessarily out of memory.
+            return SoundIoErrorNoMem;
```

Comment:
I agree with your logic here, but maybe out of distrust of Microsoft's implementation, should we perhaps check `hr` anyway and return `SoundIoErrorOpeningDevice` in case it is not `E_OUTOFMEMORY`?